src/main/java/nextstep/qna/domain/Question.java

```diff
@@ -68,43 +66,45 @@ public NsUser getWriter() {
 
     public void addAnswer(Answer answer) {
         answer.toQuestion(this);
-        answers.addAnswer(answer);
+        answers.add(answer);
     }
 
     public boolean isOwner(NsUser loginUser) {
         return writer.equals(loginUser);
     }
 
-    public void checkOwner(Question question, NsUser loginUser) throws CannotDeleteException {
-        if (!question.isOwner(loginUser)) {
-            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
-        }
-    }
-
-    public void checkAnswerOwner(NsUser loginUser) throws CannotDeleteException {
-        this.answers.checkAnswerOwner(loginUser);
-    }
-
-    public Question setDeleted(boolean deleted) {
-        this.deleted = deleted;
-        return this;
-    }
-
     public boolean isDeleted() {
         return deleted;
     }
 
-    public List<Answer> getAnswers() {
-        return answers.getAnswers();
+    public void delete(NsUser loginUser) throws CannotDeleteException {
+        if (!this.isOwner(loginUser)) {
+            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
+        }
+        for (Answer answer : answers) {
+            answer.delete(loginUser);
+        }
+        this.deleted = true;
```

Comment:
isOwner는 결국 삭제가 가능한지에 대한 체크겠죠. 지금은 isOwner하나로 해결되지만 추가적으로 기획이 들어올 수 있어요. 다음과 같이 작성해보는건 어떨까요?
```suggestion
        if (deletableQuestion(loginUser)) {
            answers.forEach(Answer::delete);
            this.deleted = true;
        }
```